torch/utils/_pytree.py

```diff
@@ -597,14 +597,32 @@ def tree_structure(tree: PyTree) -> TreeSpec:
     return tree_flatten(tree)[1]
 
 
-def tree_map(func: Callable[..., Any], tree: PyTree) -> PyTree:
-    flat_args, spec = tree_flatten(tree)
-    return tree_unflatten([func(i) for i in flat_args], spec)
+def tree_map(func: Callable[..., Any], tree: PyTree, *rest: PyTree) -> PyTree:
+    """Maps a multi-input function over pytree args to produce a new pytree.
```

Comment:
Main problem right now is that we don't want torch.utils._pytree to diverge from torch.utils._cxx_pytree. @XuehaiPan how should we proceed with this?

Reply:
Ok another datapoint then: jax also has a [`is_leaf`](https://jax.readthedocs.io/en/latest/_autosummary/jax.tree_util.tree_map.html) argument that we may want to integrate at some point (it's pretty useful to make custom PyTree structures).

If we need to adapt the cxx implementation for `*rest`, we could perhaps also integrate that extra arg there too?

Reply:
Yup

Reply:
> Main problem right now is that we don't want torch.utils._pytree to diverge from torch.utils._cxx_pytree. @XuehaiPan how should we proceed with this?

Just add `*rest` to the function signature and pass the `*rest` to `optree.tree_map` would be OK. `optree` already supports multi-tree-map.